src/app/cart/cart.component.js

```diff
@@ -97,7 +101,6 @@ class CartController {
       .configureProduct(item.code, item.config, true, item.uri)
     modal && modal.result
       .then(() => {
-        pull(this.cartData.items, item)
```

Comment:
[The original PR](https://github.com/CruGlobal/give-web/pull/338)

Reply:
OK, in that case the relevant PR is a little older: https://github.com/CruGlobal/give-web/pull/302. According to the [Jira ticket](https://jira.cru.org/browse/EP-1561), "Deleting or editing an item in the cart currently clears all the items from the display and shows the user a blank cart with a loading indicator until the cart reloaded." This PR doesn't cause a regression to that.

Reply:
So, I think the reason it was done this way was because it is mimicking how the API calls are working (delete then add). However, the load is overriding the cart in memory anyway, so it seems unnecessary as you say.